### PR TITLE
[FLINK-5047] [table] Add count sliding group-windows for batch tables

### PR DESCRIPTION
**Pull request by twalthr**
This PR adds the missing sliding group-windows for row intervals. I also added an additional table to the documentation to make it more transparent what is supported.

### CODE REVIEW

File: docs/dev/table_api.md [hunk 1]
```diff
@@ -1274,11 +1274,159 @@ A session window is defined by using the `Session` class as follows:
...
+      <td>Tumble</td>
+      <td>Event-time</td>
+      <td>Time</td>
+      <td>Both</td>
+      <td>Yes</td>
+    </tr>
+    <tr>
+      <td>Batch</td>
+      <td>Tumble</td>
+      <td>Processing-time</td>
```
Comment (fhueske) [inline docs/dev/table_api.md]: I think we should remove the processing-time batch combination. Listing it in the table makes the API look incomplete even though this combination does not make sense and will never be supported.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetSlideCountWindowAggReduceGroupFunction.scala [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.runtime.aggregate
+
+import java.lang.Iterable
+import java.util.{ArrayList => JArrayList}
+
+import org.apache.flink.api.common.functions.{CombineFunction, RichGroupReduceFunction}
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetSlideCountWindowAggReduceGroupFunction.scala]: `CombineFunction` is not used

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala [hunk 3]
```diff
@@ -354,4 +354,94 @@ class DataSetWindowAggregateITCase(
...
+      .toTable(tEnv, 'long, 'int, 'double, 'float, 'bigdec, 'string)
+
+    val windowedTable = table
+      .window(Slide over 4.rows every 2.rows on 'long as 'w)
+      .groupBy('string, 'w)
+      .select('string, 'int.count)
+
+    val expected =
+      "Hello world,2\n" +
+      "Hello,2"
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala]: shouldn't this be "Hello,4"? There are four rows with `'string = "Hello"` in the data set. Shouldn't those be aggregated in a sliding count window of size 4?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala [hunk 4]
```diff
@@ -354,4 +354,94 @@ class DataSetWindowAggregateITCase(
...
+      .fromCollection(data)
+      .toTable(tEnv, 'long, 'int, 'double, 'float, 'bigdec, 'string)
+
+    val windowedTable = table
+      .window(Slide over 2.rows every 4.rows on 'long as 'w)
+      .groupBy('string, 'w)
+      .select('string, 'int.count)
+
+    val results = windowedTable.toDataSet[Row].collect()
+    Assert.assertEquals(0, results.length)
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala]: I think we should choose a data set that actually returns a result that can be checked. An empty result could be caused by many things.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala [hunk 5]
```diff
@@ -354,4 +354,94 @@ class DataSetWindowAggregateITCase(
...
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+
+    val table = env
+      .fromCollection(data)
+      .toTable(tEnv, 'long, 'int, 'double, 'float, 'bigdec, 'string)
+
+    val windowedTable = table
+      .window(Slide over 4.rows every 2.rows on 'long as 'w)
+      .groupBy('string, 'w)
+      .select('string, 'int.count)
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala]: `count` does only verify the number of rows in a window but not which rows were aggregated. `sum` would be more meaningful, IMO.

**Comment by twalthr** [comment]
This PR has not found support by the community for quite some time. I'm not sure if it is still mergeable. Most of the functionality will also be implemented as part of Blink merging efforts. I will close the PR for now.

**Pull request closed without merging**